websocket.go

```diff
@@ -109,24 +108,30 @@ func (s *Stream) serve(ctx context.Context, requests ...models.WSRequest) (chan
 		return nil, errors.WithStack(err)
 	}
 
+	ftxChannel := models.Channel("")
+	if len(requests) > 0 {
+		ftxChannel = requests[0].Channel
+	}
+
 	doneC := make(chan struct{})
 	eventsC := make(chan interface{}, 1)
 
 	go func() {
 		go func() {
 			defer close(doneC)
+			defer close(eventsC)
 
 			for {
 				message := &models.WsResponse{}
 				err = conn.ReadJSON(&message)
 				if err != nil {
-					s.printf("read msg: %v", err)
+					s.printf("channel %v read msg: %v", ftxChannel, err)
 					if websocket.IsCloseError(err, websocket.CloseNormalClosure) {
 						return
 					}
 					conn, err = s.reconnect(ctx, requests)
```

Comment:
There's an edge case below which panics if `reconnect` returns an error, see `conn.WriteMessage` below.

Reply:
Long time since this PR, currently not working on this project, sorry for no time to deal with this.